frontend/lib/api/categories.ts

```diff
@@ -0,0 +1,13 @@
+import { z } from "zod";
+
+import { rootModelSchema } from "@/types/root";
+
+import { tagSchema } from "./tag";
+
+export const categorySchemaIntermediate = z.object({
+  name: z.string(),
+  tags: z.array(tagSchema),
+});
+
+export const categorySchema = categorySchemaIntermediate.merge(rootModelSchema);
+export type Category = z.infer<typeof categorySchema>;
```

Comment:
Are we expecting to have devs call schema.parse when they fetch results? 

We might be able to take advantage of transformResponse to have this handled in the API itself. Might be worth looking into

Reply:
Really like the use of zod here